tests/metrics/test_observers.py

```diff
@@ -0,0 +1,71 @@
+from collections import namedtuple
+
+import pandas as pd
+
+from vivarium_public_health.disease import DiseaseState, RiskAttributableDisease
+from vivarium_public_health.metrics.disability import DisabilityObserver
+
+
+def test_disability_observer_setup(mocker):
+    """Test that DisabilityObserver.setup() registers expected observations
+    and returns expected disease classes."""
+
+    observer = DisabilityObserver()
+    builder = mocker.Mock()
+    builder.results.register_observation = mocker.Mock()
+
+    # Set up fake calls for cause-specific register_observation args
+    MockCause = namedtuple("MockCause", "state_id")
+    builder.components.get_components_by_type = lambda n: [
```

Comment:
Clever. I would've stayed in mocker-land with something like `builder.components.get_components_by_type.return_value = [MockCause("flu"), MockCause("measles")]`